### PR TITLE
💄 #25 Legend Component Update

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

Updates Legend Component to use UI Kit component
Fixes menu behaviour to open on hover + 'enter' key but not on click

https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/icgc-argo/uikit/25

**Type of Change**

- [ ] Bug
- [x] Styling
- [ ] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [x] New Components with storybook stories created
  - [ ] Modified Existing components and updates stories
  - [ ] No new UIKit components or changes
- [ x Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: uikit/Legend/index.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
+import React, { useState, createRef } from 'react';
```
Comment (mistryrn) [inline uikit/Legend/index.tsx:1]: This one seems to be missing the copyright header but other than that it's all good!

**Review by mistryrn — APPROVED** [review]
Looks good, same copyright header stuff as the other PR but other than that good to merge 🚀

File: uikit/Legend/index.tsx (hunk 1 above)
Comment (demariadaniel) [reply uikit/Legend/index.tsx:1]: Updated

File: uikit/Legend/stories.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2020 The Ontario Institute for Cancer Research. All rights reserved
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply uikit/Legend/stories.tsx:2]: Updated

**Pull request merged**